### ISSUE TITLE
docker::registry is called every time

### ISSUE THREAD

**Issue by Jeoffreybauvin** [puppetlabs/puppetlabs-docker#15]
Hi there,

When using docker::registry, this resource is called at every catalogue.

A little hack : 

```  $auth_string = base64('encode', "${username}:${password}")
  if $receipt {
    # no - with pw_hash
    $local_user_strip = regsubst($local_user, '-', '', 'G')

    file { "/root/.docker/registry-auth-puppet_receipt_${server}_${local_user}":
      ensure  => $ensure,
      content => pw_hash("${title}${auth_environment}${auth_cmd}${local_user}", 'SHA-512', $local_user_strip),
      notify  => Exec["${title} auth"],
    }
  }

  exec { "${title} auth":
    environment => $auth_environment,
    command     => $auth_cmd,
    user        => $local_user,
    cwd         => '/root',
    path        => ['/bin', '/usr/bin'],
    timeout     => 0,
    refreshonly => $receipt,
  }```

**Comment by electrical** [comment]
Implemented this change at my work and solved the issue. Its to bad that the docker login process doesn't provide any functionality for checking.

**Comment by LongLiveCHIEF** [comment]
You can check the `~/.docker/config.json` for an `auths` block. Every key in this block is a registry that is logged into.  This could be a means by which the docker puppet module knows hows to execute a login.

**Comment by electrical** [comment]
@LongLiveCHIEF 
You can check that yeah but you will never be able to compare the hash with the actual details. If for example you change the login details in puppet, it won't change it in the file.

The proposed solution by TS at least allows for a comparison using the puppet file parts.
At my work we implemented it and works as expected so far.

**Comment by natemccurdy** [comment]
I've opened a ticket in Puppet's ticketing system here: https://tickets.puppetlabs.com/browse/MODULES-5988

**Comment by scotty-c** [comment]
This has been merged to master in PR #40
